packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -0,0 +1,360 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+
+import 'actions.dart';
+import 'binding.dart';
+import 'focus_manager.dart';
+import 'focus_scope.dart';
+import 'framework.dart';
+import 'inherited_notifier.dart';
+
+/// A set of [KeyboardKey]s that can be used as the keys in a [Map].
+///
+/// A key set contains the keys that are down simultaneously to represent a
+/// shortcut.
+///
+/// This is a thin wrapper around a [Set], but changes the equality comparison
+/// from an identity comparison to a contents comparison so that non-identical
+/// sets with the same keys in them will compare as equal.
+///
+/// See also:
+///
+///  - [ShortcutManager], which uses [LogicalKeySet] (a [KeySet] subclass) to
+///    define its key map.
+class KeySet<T extends KeyboardKey> extends Diagnosticable {
+  /// A constructor for making a [KeySet] of up to four keys.
+  ///
+  /// If you need a set of more than four keys, use [KeySet.fromSet].
+  ///
+  /// The `key1` parameter must not be null. The same [KeyboardKey] may
+  /// not be appear more than once in the set.
+  KeySet(
```

Comment:
Why not just have `fromSet` be the default constructor? Set literals are are pretty concise, don't have a limit of four  items. Is it that they won't assert if there are duplicates?

Reply:
Because it's just more concise and easy to read when you define them.  Compare this:
```dart
        LogicalKeySet(LogicalKeyboardKey.tab): const Intent(NextFocusAction.key),
        LogicalKeySet(LogicalKeyboardKey.shift, LogicalKeyboardKey.tab): const Intent(PreviousFocusAction.key),
```
with this:
```dart
        LogicalKeySet(<LogicalKeyboardKey>{LogicalKeyboardKey.tab}): const Intent(NextFocusAction.key),
        LogicalKeySet(<LogicalKeyboardKey>{LogicalKeyboardKey.shift, LogicalKeyboardKey.tab): const Intent(PreviousFocusAction.key}),
```

And it's not because of the duplicate checking, because Dart sets aren't multisets, so there are no duplicates in a `Set` anyhow.

The vast majority of key bindings will be single keys (or maybe two), so I thought it made sense to streamline those a bit.